tensorflow/core/kernels/mkl_qmatmul_op.cc

```diff
@@ -0,0 +1,875 @@
+// This kernel will be replaced by an MKL kernel during graph
+// optimization pass.
+REGISTER_KERNEL_BUILDER(Name("QuantizedMatMulWithBias")
+                            .Device(DEVICE_CPU)
+                            .TypeConstraint<quint8>("T1")
+                            .TypeConstraint<qint8>("T2")
+                            .TypeConstraint<qint32>("Toutput"),
+                        NoOp);
+
+REGISTER_KERNEL_BUILDER(
+    Name("_MklQuantizedMatMulWithBias")
+        .Device(DEVICE_CPU)
+        .TypeConstraint<quint8>("T1")
+        .TypeConstraint<qint8>("T2")
+        .TypeConstraint<float>("Tbias")
+        .TypeConstraint<qint32>("Toutput")
+        .Label(mkl_op_registry::kMklQuantizedOpLabel),
+    MklDnnQuantizedMatMulOp<CPUDevice, quint8, qint8, float, qint32>);
+REGISTER_KERNEL_BUILDER(
+    Name("_MklQuantizedMatMulWithBias")
+        .Device(DEVICE_CPU)
+        .TypeConstraint<quint8>("T1")
+        .TypeConstraint<qint8>("T2")
+        .TypeConstraint<qint32>("Tbias")
+        .TypeConstraint<qint32>("Toutput")
+        .Label(mkl_op_registry::kMklQuantizedOpLabel),
+    MklDnnQuantizedMatMulOp<CPUDevice, quint8, qint8, qint32, qint32>);
+
+// Register NoOp kernel for QuantizedMatMulWithBiasAndRelu to get a python
+// interface. This kernel will be replaced by an MKL kernel during
+// graph-optimization pass.
+REGISTER_KERNEL_BUILDER(Name("QuantizedMatMulWithBiasAndRelu")
+                            .Device(DEVICE_CPU)
+                            .TypeConstraint<quint8>("T1")
+                            .TypeConstraint<qint8>("T2")
+                            .TypeConstraint<qint32>("Toutput"),
+                        NoOp);
+
+// Register NoOp kernel for QuantizedIPWithBiasAndReluAndRequantize
+// to get a python interface. This kernel will be replaced by an MKL kernel
+// during graph-optimization pass.
+REGISTER_KERNEL_BUILDER(Name("QuantizedMatMulWithBiasAndReluAndRequantize")
+                            .Device(DEVICE_CPU)
+                            .TypeConstraint<quint8>("T1")
+                            .TypeConstraint<qint8>("T2")
+                            .TypeConstraint<qint32>("Tbias")
+                            .TypeConstraint<quint8>("Toutput"),
+                        NoOp);
+REGISTER_KERNEL_BUILDER(Name("QuantizedMatMulWithBiasAndReluAndRequantize")
+                            .Device(DEVICE_CPU)
+                            .TypeConstraint<quint8>("T1")
+                            .TypeConstraint<qint8>("T2")
+                            .TypeConstraint<float>("Tbias")
+                            .TypeConstraint<quint8>("Toutput"),
+                        NoOp);
+
+// Register a templatized implementation of _MklQuantizedMatMulWithBiasAndRelu.
+REGISTER_KERNEL_BUILDER(
+    Name("_MklQuantizedMatMulWithBiasAndRelu")
+        .Device(DEVICE_CPU)
+        .TypeConstraint<quint8>("T1")
+        .TypeConstraint<qint8>("T2")
+        .TypeConstraint<qint32>("Toutput")
```

Comment:
Should this have `TypeContraint` for `Tbias` as well?

Reply:
This hasn't been addressed.

Reply:
@penpornk Mistakenly, an attribute for bias type was not added in `tensorflow/core/ops/mkl_nn_ops.cc` line no 853 for this operator. I have added a comment to add that in a future PR. Can we go with current form? Because a modification here will involve modification to other places in other PRs and possibly in graph rewrite.

Reply:
Got it. Sure, let's keep it for future PRs.